File: UI/clinicnow_desktop/lib/models/report_data.dart
/// Mirrors `ClinicNow.Model.Reports.RevenueReportData` /
/// `AppointmentsReportData` - the aggregate figures behind each PDF, so the
/// report tabs can chart what the document says (review item C8) instead of
/// re-deriving it from data the client doesn't have.
class RevenueReportRow {
  final String serviceName;
  final int paymentCount;

  /// Captured minus refunded - a refunded visit isn't revenue.
  final double netTotalEur;

  const RevenueReportRow({
    required this.serviceName,
    required this.paymentCount,
    required this.netTotalEur,
  });

  factory RevenueReportRow.fromJson(Map<String, dynamic> json) => RevenueReportRow(
        serviceName: json['serviceName'] as String? ?? '',
        paymentCount: json['paymentCount'] as int? ?? 0,
        netTotalEur: (json['netTotalEur'] as num?)?.toDouble() ?? 0,
      );
}

class RevenueReportData {
  final List<RevenueReportRow> rows;
  final double grandTotalEur;

  const RevenueReportData({required this.rows, required this.grandTotalEur});

  factory RevenueReportData.fromJson(Map<String, dynamic> json) => RevenueReportData(
        rows: (json['rows'] as List<dynamic>? ?? [])
            .map((e) => RevenueReportRow.fromJson(e as Map<String, dynamic>))
            .toList(),
        grandTotalEur: (json['grandTotalEur'] as num?)?.toDouble() ?? 0,
      );
}

class AppointmentsReportRow {
  final String doctorName;
  final int pendingCount;
  final int confirmedCount;
  final int completedCount;
  final int cancelledCount;
  final int totalCount;

  const AppointmentsReportRow({
    required this.doctorName,
    required this.pendingCount,
    required this.confirmedCount,
    required this.completedCount,
    required this.cancelledCount,
    required this.totalCount,
  });

  factory AppointmentsReportRow.fromJson(Map<String, dynamic> json) => AppointmentsReportRow(
        doctorName: json['doctorName'] as String? ?? '',
        pendingCount: json['pendingCount'] as int? ?? 0,
        confirmedCount: json['confirmedCount'] as int? ?? 0,
        completedCount: json['completedCount'] as int? ?? 0,
        cancelledCount: json['cancelledCount'] as int? ?? 0,
        totalCount: json['totalCount'] as int? ?? 0,
      );
}

class AppointmentsReportData {
  final List<AppointmentsReportRow> rows;
  final int totalCount;

  const AppointmentsReportData({required this.rows, required this.totalCount});

  factory AppointmentsReportData.fromJson(Map<String, dynamic> json) => AppointmentsReportData(
        rows: (json['rows'] as List<dynamic>? ?? [])
            .map((e) => AppointmentsReportRow.fromJson(e as Map<String, dynamic>))
            .toList(),
        totalCount: json['totalCount'] as int? ?? 0,
      );
}

File: ClinicNow.Model/Reports/ReportData.cs
namespace ClinicNow.Model.Reports;

/// <summary>
/// The same figures the PDF reports print, as data (review item C8: both
/// reports need a graphical view, and a client can't chart a byte array).
///
/// Deliberately the aggregate rows only - never the underlying appointments or
/// payments. A chart needs one number per bar; shipping every row to draw a
/// dozen of them would be the "list endpoint returns heavy payloads" defect the
/// rulebook calls out (Part II §D). The PDF and these DTOs are produced by the
/// same aggregation in <c>ReportService</c>, so a chart can never disagree with
/// the document printed beside it.
/// </summary>
public class RevenueReportRow
{
    public string ServiceName { get; set; } = string.Empty;

    public int PaymentCount { get; set; }

    /// <summary>Captured minus refunded, so a refunded visit doesn't read as revenue.</summary>
    public decimal NetTotalEur { get; set; }
}

public class RevenueReportData
{
    public List<RevenueReportRow> Rows { get; set; } = [];

    public decimal GrandTotalEur { get; set; }
}

/// <summary>One doctor's appointment counts, split by status, for the period.</summary>
public class AppointmentsReportRow
{
    public string DoctorName { get; set; } = string.Empty;

    public int PendingCount { get; set; }

    public int ConfirmedCount { get; set; }

    public int CompletedCount { get; set; }

    public int CancelledCount { get; set; }

    public int TotalCount { get; set; }
}

public class AppointmentsReportData
{
    public List<AppointmentsReportRow> Rows { get; set; } = [];

    public int TotalCount { get; set; }
}
